### PR TITLE
RAT tests

### LINKED ISSUE #475: Add tests for RAT
There is no unit tests for RAT class (coreblocks/struct_common/rat.py). Please add them.

### PR DESCRIPTION
**Pull request by JumiDeluxe**
Basic unit tests for FRAT and RRAT structs
`2000` steps

Resolves #475

### CODE REVIEW

File: test/structs_common/test_rat.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+from coreblocks.params.configurations import test_core_config
+
+from collections import deque
+from random import Random
+
+
+class TestFrontendRegisterAliasTable(TestCaseWithSimulator):
+    def gen_input(self):
+        while len(self.expected_entries) < self.log_regs:
+            yield
```
Comment (tilk) [inline test/structs_common/test_rat.py:14]: Dead code

File: test/structs_common/test_rat.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+from coreblocks.structs_common.rat import FRAT, RRAT
+from coreblocks.params import GenParams
+from coreblocks.params.configurations import test_core_config
+
+from collections import deque
+from random import Random
+
+
+class TestFrontendRegisterAliasTable(TestCaseWithSimulator):
+    def gen_input(self):
```
Comment (tilk) [inline test/structs_common/test_rat.py:12]: Not a process, should be run before the test starts.

File: test/structs_common/test_rat.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            rp = self.rand.randrange(1, 2**self.gen_params.phys_regs_bits) if rl != 0 else 0
+            rl_s1 = self.rand.randrange(self.gen_params.isa.reg_cnt)
+            rl_s2 = self.rand.randrange(self.gen_params.isa.reg_cnt)
+
+            self.to_execute_list.append({"rl": rl, "rp": rp, "rl_s1": rl_s1, "rl_s2": rl_s2})
+
+    def do_rename(self):
+        for _ in range(self.test_steps):
+            while len(self.to_execute_list) == 0:
+                yield
```
Comment (tilk) [inline test/structs_common/test_rat.py:26]: If the above will be run before simulation start, this won't be needed.

Comment (JumiDeluxe) [reply test/structs_common/test_rat.py]: Removed

File: test/structs_common/test_rat.py (hunk 1 above)
Comment (JumiDeluxe) [reply test/structs_common/test_rat.py]: Removed

**Review by tilk — APPROVED** [review]
Looks okay.

**Review by lekcyjna123 — APPROVED** [review]

**Pull request merged**